integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java

```diff
@@ -16,58 +16,149 @@
 
 package io.helidon.integrations.vault.secrets.pki;
 
+import java.util.function.Function;
+
+import io.helidon.common.http.Http;
+import io.helidon.integrations.common.rest.RestApi;
 import io.helidon.integrations.vault.ListSecrets;
+import io.helidon.integrations.vault.Vault;
 import io.helidon.integrations.vault.VaultOptionalResponse;
+import io.helidon.integrations.vault.VaultRestException;
+
+import jakarta.json.JsonObject;
 
 class PkiSecretsImpl implements PkiSecrets {
-    private final PkiSecretsRx delegate;
+    private final RestApi restApi;
+    private final String mount;
 
-    PkiSecretsImpl(PkiSecretsRx delegate) {
-        this.delegate = delegate;
+    PkiSecretsImpl(RestApi restApi, String mount) {
+        this.restApi = restApi;
+        this.mount = mount;
     }
 
     @Override
     public VaultOptionalResponse<ListSecrets.Response> list(ListSecrets.Request request) {
-        return delegate.list(request).await();
+        String apiPath = mount + "/certs";
+
+        return restApi.invokeOptional(Vault.LIST,
+                                      apiPath,
+                                      request,
+                                      VaultOptionalResponse.<ListSecrets.Response, JsonObject>vaultResponseBuilder()
+                                              .entityProcessor(ListSecrets.Response::create));
     }
 
     @Override
     public CaCertificateGet.Response caCertificate(CaCertificateGet.Request request) {
-        return delegate.caCertificate(request).await();
+        String apiPath = switch (request.format()) {
+            case DER -> mount + "/ca";
+            case PEM, PEM_BUNDLE -> mount + "/ca_chain";
+        };
+
+        VaultOptionalResponse<byte[]> response = restApi.getBytes(apiPath,
+                                                                  request,
+                                                                  VaultOptionalResponse.<byte[], byte[]>vaultResponseBuilder()
+                                                                          .entityProcessor(Function.identity()));
+        return CaCertificateGet.Response.builder()
+                .entity(response.entity().orElseThrow(() ->
+                                                              VaultRestException.builder()
```

Comment:
Indentation seems excessive, can this be aligned with `response.entity` ?
I.e.

```java
.entity(response.entity().orElseThrow(() ->
        VaultRestException.builder()
                          .headers(response.headers())
                          .requestId(response.requestId())
                          .status(response.status())
                          .vaultErrors(response.errors())
                          .message(
                                  "CA Certificate is expected to be present, but is"
                                          + " not available")
                          .build()))
```

Reply:
This is how the idea template is updating it so will touch base on this.

Reply:
I tried updating it manually, but then doing reformat is putting it at same location again.

Reply:
You can actually do it this way:

```java
return CaCertificateGet.Response.builder()
        .entity(response.entity()
                        .orElseThrow(() -> VaultRestException.builder()
                                .headers(response.headers())
                                .requestId(response.requestId())
                                .status(response.status())
                                .vaultErrors(response.errors())
                                .message(
                                        "CA Certificate is expected to be present, but is"
                                                + " not available")
                                .build()))
        .headers(response.headers())
        .requestId(response.requestId())
        .status(response.status())
        .build();
```

Reply:
updated it.